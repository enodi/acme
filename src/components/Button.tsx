import React from "react";

import { ReactComponent as Chat } from "../assets/chat.svg";

interface ButtonProps {
  text?: string;
}

const Button: React.FC<ButtonProps> = ({ text }) => (
  <button className={text ? "button-text" : "button-img"}>
    {text ? text : <Chat style={{ width: 18, height: 18 }} fill={"#fff"} />}
  </button>
);

export default Button;
