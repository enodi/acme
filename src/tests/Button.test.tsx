import React from "react";
import { shallow } from "enzyme";

import Button from "../components/Button";

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<Button text="Follow up" />);
});

test("should render Button component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
