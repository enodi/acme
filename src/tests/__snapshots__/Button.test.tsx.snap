// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component correctly 1`] = `
<button
  className="button-text"
>
  Follow up
</button>
`;
